### ISSUE TITLE
create_app may throw unexpectedly

### ISSUE THREAD

**Issue by atzannes** [martyzz1/heroku3.py#37]
https://github.com/martyzz1/heroku3.py/blob/7349e31a2255a4868b42992e604d0bcfd9006fbf/heroku3/api.py#L299 : `app = self.app(name)` may throw unexpectedly: `requests.exceptions.HTTPError: 403 Client Error: Forbidden for url: xxx`

In the except clause above, we are trying to recover from an exception when the app name requested is already in use. The problem occurs when the app name requested is in use by a different heroku user. App names are required to be unique across users, presumably because the app URL needs to be unique. In that scenario, an exception `422 Client Error: {"id":"invalid_params","message":"Name is already taken"}` is thrown, converted into a warning and we attempt to return a reference to the app. This works if the app is owned by the same user that is trying to create the new app with the existing name, but fails otherwise, in which case, the error/exception is a bit cryptic.

A solution could be:
```
        except HTTPError as e:
            saved_exc = (sys.exc_type, sys.exc_value, sys.exc_traceback)
            if "Name is already taken" in str(e):
                try:
                    app = self.app(name)
                    print("Warning - {0:s}".format(e))
                except:
                    raise saved_exc[0], saved_exc[1], saved_exc[2]
            else:
                raise saved_exc[0], saved_exc[1], saved_exc[2]
```
Using `raise saved_exc[0], saved_exc[1], saved_exc[2]` instead of `raise e` re-raises the exception as if it had not been handled, that way the stack-trace is not modified and it likely results in it being more informative. Saving the exception values in `saved_exc` is necessary; we cannot simply use `raise` without an argument, because if the line `app = self.app(name)` raises and we subsequently use `raise` without an argument, we will re-raise that inner exception.

**Comment by martyzz1** [comment]
I'd happily take a Pull request on this...

**Comment by atzannes** [comment]
I tried to push to a branch to create a pull request, but I don't seem to have access rights

**Comment by martyzz1** [comment]
you need to fork the project in github....  Then clone your fork, and push your changes to your master branch.... (think of your forked project's master branch as a branch of mine)

Then raise a pull request...

**Comment by StarkBotsIndustries** [comment]
Something like this happened to me just now when I tried calling `create_app`. It throws the error `requests.exceptions.HTTPError: 403 Client Error: Forbidden for url: https://api.heroku.com/apps`. I tried to get other error like
1) Using Other app's name which should give : `422 Client Error: {"id":"invalid_params","message":"Name is already taken"}` but it threw the _same error_
2) Using New name but it didn't create new app just threw the _same error_.
3) Then I tried replacing the value of app name, stack, region with None but this time it **created the app**.
4) I tried replacing one or two of them with None and it threw the _same error_.
5) This time I tried it in a different file and it worked at third attempt when I used keyword arguments and it worked later whenever I tried in it. That's weird and random too.

So this was my weird approach to make things work and I didn't get a solution. Sorry for taking your time but am I missing something ?
